core/src/br/cefetmg/games/minigames/ToothRunner.java

```diff
@@ -0,0 +1,247 @@
+            this.sobe = this.impulso;
+
+            TextureRegion[][] quadrosDaAnimacao;
+            quadrosDaAnimacao = TextureRegion.split(textura, 610, 755);
+
+            corre = new Animation(1f, new TextureRegion[] {
+                    quadrosDaAnimacao[0][0],
+                    quadrosDaAnimacao[0][1],
+                    quadrosDaAnimacao[0][2],
+            });
+
+            corre.setPlayMode(Animation.PlayMode.LOOP_PINGPONG);
+            tempoAnimacao = 0;
+        }
+    }
+
+    private class Enemie {
+        public Sprite sprite;
+        public Vector2 direcaoX;
+
+        public Enemie(Texture textura, Vector2 direcaoX) {
+            this.direcaoX = direcaoX;
+            this.sprite = new Sprite(textura);
+        }
+    };
+
+    private final Dente dente;
+    private ArrayList<Enemie> pirulito;
+
+    private final Texture pirulitoTexture;
+    private final Texture denteAnimation;
+    private final Sound passo;
+    private final Sound pulo;
+
+    private int velocidade;
+    private Vector2 gravidade;
+    private Random random;
+    private int delta;
+
+    private Texture bg;
+
+    public ToothRunner(BaseScreen screen, GameStateObserver observer, float difficulty) {
+        super(screen, difficulty, 10000, TimeoutBehavior.WINS_WHEN_MINIGAME_ENDS, observer);
+
+        delta = 0;
+        this.pirulito = new ArrayList<Enemie>();
+        this.pirulitoTexture = this.screen.assets.get("tooth-runner/pirulito.png", Texture.class);
+        this.denteAnimation = this.screen.assets.get("tooth-runner/denteAnimacao.png", Texture.class);
+        this.passo = this.screen.assets.get("tooth-runner/passo.mp3", Sound.class);
+        this.pulo = this.screen.assets.get("tooth-runner/pulo.mp3", Sound.class);
+
+        this.gravidade = new Vector2(0, (float) 1);
+        this.random = new Random();
+
+        bg = new Texture(Gdx.files.internal("tooth-runner/fundo.png"));
+        bg.setWrap(Texture.TextureWrap.Repeat, Texture.TextureWrap.Repeat);
+        this.dente = new Dente(denteAnimation);
+        this.dente.sprite.setSize(100, 100);
+        this.dente.sprite.setX(100);
+        this.dente.sprite.setY(100);
+
+        createEnemies();
+    }
+
+    void createEnemies() {
+        int distancia = 0;
+        for (int i = 0; i < 3; i++) {
+            Enemie aux;
+            aux = new Enemie(pirulitoTexture, new Vector2(-1, 0));
+            aux.sprite.setSize(100, 100 + random.nextInt(30));
+            aux.sprite.setX(this.screen.viewport.getWorldWidth()-aux.sprite.getWidth() + distancia);
+            distancia += this.screen.viewport.getWorldWidth() / 3 + 100;
+            aux.sprite.setY(100);
+            this.pirulito.add(aux);
+        }
+    }
+
+    @Override
+    protected void configureDifficultyParameters(float difficulty) {
+        this.velocidade = 8 + (int)(difficulty*6);
+    }
+
+    @Override
+    public void onHandlePlayingInput() {
+        Gdx.input.setInputProcessor(
```

Comment:
Aqui está acontecendo um problemão de desempenho (a máquina virtual do java estourou o espaço do _heap_ aqui):

- O método `onHandlePlayingInput()` é invocado dentro do `update()`, ou seja, a cada quadro
- Aqui dentro, seu código está instanciando um novo objeto que implementa a interface `InputProcessor` e instanciar novos objetos a cada quadro é péssimo para o desempenho. Tem casos que precisamos instanciar a cada quadro, mas neste não é necessário.
- Para corrigir, há pelo menos duas opções:
  1. (**minha sugestão**) Em vez de definir um processador de entradas, use apenas `if (Gdx.input.justTouched()) { /* ... */ }`
    - Esta é a melhor alternativa, visto que vocês escreverão bem menos código e não instanciarão nenhum objeto na memória desnecessariamente
  1.  Chamar `Gdx.input.setInputProcessor()` dentro do método `create` (que é chamado apenas 1x
    - Além dessa alteração, também será necessário chamar `Gdx.input.setInputProcessor(null)` ao final do MiniGame para que este processador de inputs seja "desligado" quando ele for encerrado